src/catalog/catalog.cpp

```diff
@@ -78,26 +118,28 @@ table_oid_t Catalog::CreateUserTable(transaction::TransactionContext *txn, db_oi
   auto table_handle = ns_handle.GetTableHandle(txn, ns_oid);
 
   // creates the storage table and adds to pg_class
-  auto tbl_rw = table_handle.CreateTable(txn, schema, table_name);
+  auto tbl_rw = table_handle.CreateUserTable(txn, schema, table_name);
+
+  // Modification to support indexes
+  table_index_map_.emplace(tbl_rw->Oid(), std::unordered_set<index_oid_t>());
```

Comment:
I don't know how this map is being used (or why we need it), but I feel like it's going to cause problems later on because it's not transactional.